google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/it/BuiltinMetricsIT.java

```diff
@@ -176,5 +329,63 @@ private void verifyMetricsArePublished(
     assertWithMessage("View " + view + " didn't return any data.")
         .that(response.getTimeSeriesCount())
         .isGreaterThan(0);
+
+    // Compare metric data with in memory metrics reader data if present
+    if (dataFromReader != null) {
```

Comment:
How can we make this more deterministic? its very unclear under what circumstances this will be present and how do we know if thats not a test failure?

Reply:
THis is confusing, split it into 2 methods.